### PR TITLE
[WIP] Bike racks

### PR DESCRIPTION
**Pull request by mlangsdorf**
C++ is hard.

### CODE REVIEW

File: src/vehicle.cpp [hunk 1]
```diff
@@ -6819,3 +6853,135 @@ void vehicle::use_monster_capture( int part, const tripoint &pos )
...
+        }
+        if( parts[ possible_part ].mount.y == parts[ part ].mount.y ) {
+            y_parts.push_back( possible_part );
+        }
+    }
+    std::vector<std::vector<int>> ret_parts;
+    if( x_parts.size() > 1 ) {
+        std::vector<int> x_ret;
+        // sort by Y-axis, since they're all on the same X-axis
+        std::sort( x_parts.begin(), x_parts.end(), [&]( const int const *lhs, const int const *rhs ) {
```
Comment (OrenAudeles) [inline src/vehicle.cpp]: try `[&]( const int lhs, const int rhs ){ ... }`

File: src/vehicle.cpp [hunk 2]
```diff
@@ -6819,3 +6853,135 @@ void vehicle::use_monster_capture( int part, const tripoint &pos )
...
+            }
+        }
+        std::swap_ranges( x_parts.begin() + i, x_parts.begin() + last_p, x_ret.begin() );
+        ret_parts.push_back( x_ret );
+    } else {
+        ret_parts.push_back( x_parts );
+    }
+    if( y_parts.size() > 1 ) {
+        std::vector<int> y_ret;
+        const auto sorter = [&]( const int *lhs, const int *rhs ) {
```
Comment (OrenAudeles) [inline src/vehicle.cpp]: try `[&]( const int lhs, const int rhs ){ return ( parts[lhs].mount.x > parts[rhs].mount.x ); }`

**Comment by mlangsdorf** [comment]
Todo:
1. Use coord_translate( x_part.mount + delta ) instead of global_part_pos3
2. simplify target vehicle logic - do all the checks at the end.

**Comment by mlangsdorf** [comment]
figure out merge co-ordinates.

**Comment by mlangsdorf** [comment]
mathematically reversing tileray operations is way huge pain in the ass.
but we know the bike rack part mounts and global pos3 and the target vehicle global pos3. the merge vehicle has to be cardinally adjacent to the bike rack. calculate global pos3 for cardinally adjacent mount points, use the mount point that has the same global pos3 as the target vehicle.

**Comment by mlangsdorf** [comment]
Current status:
1. Merge works. Sometimes. 
2. Merged vehicles aren't 100% correctly disabled (needs support in pickup for ignoring disabled parts)
3. Need to add split code.
4. Bike racks aren't detecting that they're full correctly.

**Comment by mlangsdorf** [comment]
New vague plan: store carried vehicle X or Y offset (it's one or the other, by definition) and just call the pivot point X0 or Y0 for vehicles with an X axis or Y axis. Then don't recalculate positions, just pull them off.

**Pull request closed without merging**